pkg/apps/application-list.jsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+
+import PackageKit from "./packagekit";
+import { left_click, icon_url, show_error, launch, ProgressBar, CancelButton } from  "./utils.jsx";
+
+const _ = cockpit.gettext;
+
+class ApplicationRow extends React.Component {
+    constructor() {
+        super();
+        this.state = { progress: false };
+    }
+
+    render() {
+        var self = this;
+        var comp = self.props.comp;
+        var state = self.state;
+
+        function navigate() {
+            cockpit.location.go(comp.id);
+        }
```

Comment:
One-liner which is being called just once, and the `navigate` name is being used in another function (although in a different file); I'd just fold that into the caller.

Reply:
Done.